core/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightFieldTests.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.fetch.subphase.highlight;
+
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.text.Text;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode;
+
+public class HighlightFieldTests extends ESTestCase {
+
+    public HighlightField createTestItem() {
+        String name = frequently() ? randomAsciiOfLengthBetween(1, 20) : randomRealisticUnicodeOfCodepointLengthBetween(1, 20);
+        Text[] fragments = null;
+        if (frequently()) {
+            int size = randomIntBetween(0, 5);
+            fragments = new Text[size];
+            for (int i = 0; i < size; i++) {
+                fragments[i] = new Text(
+                        frequently() ? randomAsciiOfLengthBetween(10, 30) : randomRealisticUnicodeOfCodepointLengthBetween(10, 30));
+            }
+        }
+        return new HighlightField(name, fragments);
+    }
+
+    public void testFromXContent() throws IOException {
+        HighlightField highlightField = createTestItem();
+        XContentBuilder builder = XContentFactory.contentBuilder(randomFrom(XContentType.values()));
+        if (randomBoolean()) {
+            builder.prettyPrint();
+        }
+        builder.startObject();  // we need to wrap xContent output in proper object to be able to parse it
+        builder = highlightField.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        builder.endObject();
+        XContentParser parser = XContentFactory.xContent(builder.bytes()).createParser(builder.bytes());
+        HighlightField parsedField = HighlightField.fromXContent(parser);
+        assertEquals(highlightField.getName(), parsedField.getName());
+        assertArrayEquals(highlightField.getFragments(), parsedField.getFragments());
+        if (highlightField.fragments() != null) {
+            assertEquals(XContentParser.Token.END_ARRAY, parser.currentToken());
+        }
+        assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+        assertNull(parser.nextToken());
+    }
+
+    public void testFromXContentErrors() throws IOException {
+        {
```

Comment:
Good point. What about versions? If we send a request to an incompatible backend, the request is still valid for some reason but the response format has changed. We might want to detect that. But I agree it is probably not necessary atm.